src/JWK.js

```diff
@@ -225,6 +227,100 @@ class JWK {
     let { alg, cryptoKey } = this
     return JWA.decrypt(alg, cryptoKey, ciphertext, iv, tag, aad)
   }
+
+  /**
+   * thumbprint
+   *
+   * @description
+   * Calculate the SHA-256 JWK Thumbprint according to [RFC7638]{@link https://tools.ietf.org/html/rfc7638}.
+   * This method is used to create a unique `kid` if none is specified.
+   *
+   * @example <caption>SHA-256 Thumbprint</caption>
+   * jwk.thumbprint()
+   * //
+   * // (line breaks for display only)
+   * //
+   * // => "45BLsBiWcghaEf_NF70Gf5oQcYLHaA
+   * //     tks0C48tT5SJ4"
+   *
+   * @return {String} JWK Thumbprint String
+   */
+  thumbprint () {
+    let { kty } = this
+    let data
+
+    // RSA JWK Thumbprint Fields
+    if (kty === 'RSA') {
+      let { e, n } = this
+      data = { e, kty, n }
+
+    // ECDSA JWK Thumbprint Fields
+    } else if (kty === 'EC') {
+      let { crv, x, y } = this
+      data = { crv, kty, x, y }
+
+    // Symmetric JWK Thumbprint Fields
+    } else if (kty === 'oct') {
+      let { k } = this
+      data = { k, kty }
+
+    // Invalid kty
+    } else {
+      throw new DataError('Invalid \'kty\'')
+    }
+
+    let hash = crypto.createHash('sha256')
```

Comment:
Why usage of `crypto` here, instead of `@trust/webcrypto`? This'll make it impossible to Webpack.

Reply:
Good point. Will update shortly.

Reply:
Updated :+1: